SimTracker/TrackTriggerAssociation/plugins/TTClusterAssociator.cc

```diff
@@ -183,15 +172,12 @@ void TTClusterAssociator<Ref_Phase2TrackerDigi_>::produce(edm::Event& iEvent, co
 
     /// Clean the maps that need cleaning
     /// Prepare the output map wrt TrackingParticle
-    std::map<edm::Ptr<TrackingParticle>,
-             std::vector<edm::Ref<edmNew::DetSetVector<TTCluster<Ref_Phase2TrackerDigi_>>,
-                                  TTCluster<Ref_Phase2TrackerDigi_>>>>::iterator iterMapToClean;
+    std::map<TrackingParticlePtr, std::vector<TTClusterRef>>::iterator iterMapToClean;
     for (iterMapToClean = trackingParticleToClusterVectorMap.begin();
          iterMapToClean != trackingParticleToClusterVectorMap.end();
          ++iterMapToClean) {
       /// Get the vector of references to TTCluster
-      std::vector<edm::Ref<edmNew::DetSetVector<TTCluster<Ref_Phase2TrackerDigi_>>, TTCluster<Ref_Phase2TrackerDigi_>>>
-          tempVector = iterMapToClean->second;
+      std::vector<TTClusterRef> tempVector = iterMapToClean->second;
```

Comment:
avoid copy and reassignment: `std::vector<TTClusterRef>& tempVector = iterMapToClean->second;`
(then delete line `iterMapToClean->second = tempVector;` below)

Reply:
done